packages/mui-material/src/Alert/Alert.d.ts

```diff
@@ -44,6 +51,11 @@ export interface AlertProps extends StandardProps<PaperProps, 'variant'> {
    * Set to `false` to remove the `icon`.
    */
   icon?: React.ReactNode;
+  /**
+   * The icon to display in place of the default close icon.
+   * @default <CloseIcon fontSize="small" />
+   */
+  closeIcon?: React.ReactNode;
```

Comment:
Can you use `components` prop API (`components.closeIcon`) like we have for other components? Example [Badge](https://mui.com/material-ui/api/badge/). This is for consistency with API names and ability to add multiple slots within `components` prop in future.

Reply:
I can definitely make this change. My rationale for doing it this way was to stay in line with the `icon` prop above. I'm assuming we're going to want to keep the existing component props as they are instead of introducing a breaking change to include them in a `components` prop, is that correct?

Reply:
After investigating this more, I'm not sure this is actually what we want. The `components` prop seems to take in a string ('div', 'span', 'section', etc...) and pass that along to an `as` prop somewhere along the way.

This change is more akin to the `startIcon` and `endIcon` on [Button](https://mui.com/material-ui/api/button/), where the entire node is being switched out, props and all. I'm starting to feel like `components` and `componentsProps` may not be the pattern to use here -  let me know what you think.

Reply:
> I'm assuming we're going to want to keep the existing component props as they are instead of introducing a breaking change to include them in a components prop, is that correct?

Yes correct. Moving it now would be a breaking change. In v6, we might remove it and move it under `components` prop having an `icon` slot.

> After investigating this more, I'm not sure this is actually what we want. The `components` prop seems to take in a string ('div', 'span', 'section', etc...) and pass that along to an `as` prop somewhere along the way.

I think you misread it. It is the `component` prop not `components` prop that does it. See `component` prop in [Button API](https://mui.com/material-ui/api/button/) which uses the `as` prop.
 
> This change is more akin to the `startIcon` and `endIcon` on [Button](https://mui.com/material-ui/api/button/), where the entire node is being switched out, props and all. I'm starting to feel like `components` and `componentsProps` may not be the pattern to use here - let me know what you think.

You can switch the entire node out with the `components` prop API and pass it's corresponding props with `componentsProps`:

```jsx
<Alert
  components={{ CloseIcon: MyCloseIcon }}
  componentsProps={{ closeIcon:  { fontSize: 'big' } }}
/>;
```

Then Material UI internally handles it with:

```jsx
const CloseIcon = componentsProps.CloseIcon;

// render it internally in Alert.js as
<CloseIcon {...componentsProps.closeIcon}/> ?? <CloseIcon fontSize="small" />
```

The disadvantage of passing in a custom React node directly as a prop (example below) is it causes the component to re-render as it is new instance for React everytime in each render.
```jsx
<Alert
  components={{ CloseIcon: <MyCloseIcon /> }}
  // or
  closeIcon={<MyCloseIcon />}
/>;
```

More discussion about the `components` and `componentsProps` pattern is done in #21453. It is also used as a pattern in MUI Base.